src/os/bluestore/BlueStore.cc

```diff
@@ -4303,6 +4334,23 @@ void BlueStore::_set_blob_size()
            << std::dec << dendl;
 }
 
+void BlueStore::_update_osd_memory_target_size()
+{
+  uint64_t new_osd_memory_target = cct->_conf.get_val<Option::size_t>("osd_memory_target");
+
+  if (new_osd_memory_target <= 0 ||
```

Comment:
I think this isn't a perfect place for such checks - we don't have any means to return errors while logging an error is rather invisible to user. Moreover this logic (which values are allowed and how do they relate) belongs to pcm. Hence it should be encapsulated there if any.
So right place is either top-level configuration interface or pcm. In the latter case it's preferably to have some means to report errors clearly for users.

Reply:
I have used the top-level configuration interface to check for the validity of the config options set by the user. This layer provides the mechanism to return an error if the validation fails. Returning errors from the pcm doesn't appear to be trivial and so I have gone with the former approach. I have set a minimum value for each of the config options. If the config value is below this minimum, the interface propagates the error back to the user.